cli/src/opts/cast.rs

```diff
@@ -285,12 +285,14 @@ Examples:
     Block {
         #[clap(
             help = "The block height you want to query at.",
-            long_help = "The block height you want to query at. Can also be the tags earliest, latest, or pending.",
+            long_help = "The block height you want to query at. Can also be the tags earliest, finalized, safe, latest, or pending.",
             value_name = "BLOCK"
         )]
-        block: BlockId,
+        block: Option<BlockId>,
 
         #[clap(
+            long,
+            short = 'f',
```

Comment:
this is a breaking change.

is this necessary for this PR?

Reply:
based on my understanding of the [comment](https://github.com/foundry-rs/book/pull/851#discussion_r1139371477) that started the original issue, commands that query the info of the block itself (which requires only the `BlockId` as input) should take `block` as an (optional) argument, instead of an option, and default to `latest` if not specified, e.g. `cast age` and `cast basefee`

imho, `cast block` also falls in that category (querying the info of the block itself), so we should follow those 2 commands and make `block` an optional argument. but then changing `field` to be an option is required to avoid ambiguity (having 2 optional arguments, `block` and `field`). (also, "semantic" wise, `cast block [BLOCK] -f <FIELD>`  seems more reasonable to me then the current `cast block <BLOCK> [FIELD]`, since `field`  is just for filtering the block info output)

please let me know what you think and your suggestion on this

Reply:
thank you for this,

this makes sense. 

so this is now necessary because block is now optional and defaults to latest?

I agree

> cast block [BLOCK] -f <FIELD>

is the best syntax, do we need to update the book reference?

Reply:
> so this is now necessary because block is now optional and defaults to latest?

yes

> do we need to update the book reference?

i'll update the book reference accordingly once this pr is set to be merged